### PR TITLE
Order candidates, add precision and timestamp; add precinct scraping support

### PR DESCRIPTION
**Pull request by hancush**
# Description

See title! N.b., the precinct scrape takes about a minute, as those pages are slow to load.

## Testing instructions

- Run `python -m election_night.get_results --precinct --election="2023 Municipal General - 2/28/2023"` and confirm that results are ordered, percentages have two decimal points, the timestamp is added to all files, and spot check that results match those reported for the February General Election on the BoE website: https://chicagoelections.gov/en/election-results.asp?election=241&race=11
- Run `python -m election_night.get_results --test` and confirm you see results, all 0s, for the runoff
- Run `python -m election_night.get_results --precinct` and confirm you see results, all 0s, for the runoff

### CODE REVIEW

File: election_night/get_results.py [hunk 1]
```diff
@@ -4,6 +4,129 @@
...
+
+    def parse_result(self):
+        raise NotImplementedError
+
+    def write_report(self, race_name, results):
+        with open(os.path.join(self.output_directory, f"{race_name}.csv"), "w") as output_file:
+            writer = csv.DictWriter(
+                output_file,
+                fieldnames=["Candidate", "Votes Total", "Votes Percent"],
+                extrasaction="ignore"
```
Comment (hancush) [inline election_night/get_results.py:40]: TIL! This ignores any data keys that you don't pass in as a fieldname.

**Review by derekeder — CHANGES_REQUESTED** [review]
@hancush getting an error when running:

```bash
(elections) derekeder@Rhodes block-club-election-night % python -m election_night.get_results --precinct --election="2023 Municipal General - 2/28/2023"
Getting precinct results for 2023 Municipal General - 2/28/2023! 🎉
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/derekeder/projects/datamade/block-club-election-night/election_night/get_results.py", line 161, in <module>
    PrecinctClient()[options.election]
KeyError: '2023 Municipal General - 2/28/2023'
```

getting a similar error when running:

```bash
(elections) derekeder@Rhodes block-club-election-night % python -m election_night.get_results --precinct
Getting precinct results for 2023 Municipal Runoffs - 4/4/23! 🎉
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/derekeder/projects/datamade/block-club-election-night/election_night/get_results.py", line 161, in <module>
    PrecinctClient()[options.election]
KeyError: '2023 Municipal Runoffs - 4/4/23'
```

**Comment by hancush** [comment]
@derekeder Whoops, forgot to instruct you to upgrade your requirements. Run this, then the test commands: `pip install --upgrade --no-cache-dir -r requirements.txt`

**Review by derekeder — APPROVED** [review]
@hancush ok after deleting my virtual environment and recreating and installing from scratch, the scraper commands ran as expected.

my only note/suggestion would be - if we're giving the ability to choose which election to scrape, it would be helpful to use that to name the folder the results go in. i ran both the runoff and the feb 28 back to back and i had to inspect the files in each folder to know which was which.

**Pull request merged**